packages/react-meteor-data/useTracker.js

```diff
@@ -0,0 +1,188 @@
+            );
+          }
+          refs.computationCleanup = cleanupHandler;
+        }
+      }
+      // This will capture data synchronously on first run (and after deps change).
+      // Don't run if refs.isMounted === false. Do run if === null, because that's the first run.
+      if (refs.isMounted === false) {
+        return;
+      }
+      if (refs.isMounted === null) {
+        refs.isMounted = false;
+      }
+      runReactiveFn(c);
+    } else {
+      // If deps are anything other than an array, stop computation and let next render handle reactiveFn.
+      // These null and undefined checks are optimizations to avoid calling Array.isArray in these cases.
+      if (deps === null || deps === undefined || !Array.isArray(deps)) {
+        dispose();
+      } else if (refs.isMounted) {
+        // Only run the reactiveFn if the component is mounted.
+        runReactiveFn(c);
+      } else {
+        // If not mounted, defer render until mounted.
+        refs.doDeferredRender = true;
+      }
+      // :???: I'm not sure what would happen if we try to update state after a render has been tossed - does
+      // it throw an error? It seems to cause no problems to call forceUpdate on unmounted components.
+      forceUpdate();
+    }
+  };
+
+  // We are abusing useMemo a little bit, using it for it's deps compare, but not for it's memoization.
+  useMemo(() => {
+    // if we are re-creating the computation, we need to stop the old one.
+    dispose();
+
+    // When rendering on the server, we don't want to use the Tracker.
+    if (Meteor.isServer) {
+      refs.computation = null;
+      tracked(null);
+    } else {
+      // Use Tracker.nonreactive in case we are inside a Tracker Computation.
+      // This can happen if someone calls `ReactDOM.render` inside a Computation.
+      // In that case, we want to opt out of the normal behavior of nested
+      // Computations, where if the outer one is invalidated or stopped,
+      // it stops the inner one.
+      refs.computation = Tracker.nonreactive(() => Tracker.autorun(tracked));
+
+      // We are creating a side effect in render, which can be problematic in some cases, such as
+      // Suspense or concurrent rendering or if an error is thrown and handled by an error boundary.
+      // We still want synchronous rendering for a number of reason (see readme), so we work around
+      // possible memory/resource leaks by setting a time out to automatically clean everything up,
+      // and watching a set of references to make sure everything is choreographed correctly.
+      if (!refs.isMounted) {
+        refs.disposeId = setTimeout(() => {
+          if (!refs.isMounted) {
+            dispose();
+          }
+        }, 50);
+      }
+    }
+  }, deps);
+
+  useEffect(() => {
+    // Now that we are mounted, we can set the flag, and cancel the timeout
+    refs.isMounted = true;
+
+    if (!Meteor.isServer) {
+      clearTimeout(refs.disposeId);
```

Comment:
minor note, it's cleaner to check `refs.disposeId` and if given `clearTimeout`

Reply:
At this point in the execution there will always be a `disposeId`. It always creates a timeout on first render (onBeforeMount), then the `useEffect` runs at some point after that (or never if the render is tossed), so there's always a `disposeId` to cancel.